globus_automate_client/cli/rich_helpers.py

```diff
@@ -0,0 +1,435 @@
+from .constants import OutputFormat
+from .helpers import get_http_details
+from .rich_rendering import cli_content
+
+
+def humanize_datetimes(dt: str) -> str:
+    return arrow.get(dt).humanize()
+
+
+def humanize_auth_urn(urn: str) -> str:
+    id = urn.split(":")[-1]
+    if urn.startswith("urn:globus:auth:identity:"):
+        return f"User: {id}"
+    elif urn.startswith("urn:globus:groups:id:"):
+        return f"Group: {id}"
+    return urn
+
+
+def identity_to_user(field: str, l: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    """
+    Given a list of dict entries, this function will attempt to
+    """
+    # get_identities will fail if there's no data, so short circuit
+    if len(l) == 0:
+        return l
+
+    # Only do the conversion if the user is already logged in
+    authzs = get_authorizers_for_scopes(["openid"], no_login=True)
+    authorizer = authzs.get("openid")
+    if authorizer is None:
+        return l
+
+    # Collect IDs from the List data
+    creators: Dict[str, None] = collections.OrderedDict()
+    for item in l:
+        urn_id = item.get(field, "")
+        id = urn_id.split(":")[-1]
+        creators[id] = None
+
+    # Get id -> username mapping
+    ac = AuthClient(authorizer=authorizer)
+    resp = ac.get_identities(ids=creators.keys())
+    id_to_user = {i["id"]: i["username"] for i in resp.data["identities"]}
+
+    # Update items in list
+    for item in l:
+        urn_id = item.get(field, "")
+        id = urn_id.split(":")[-1]
+        if id in id_to_user:
+            item[field] = id_to_user[id]
+    return l
+
+
+class Field:
+    """
+    A generic class structure for transforming lists of data into a table. Each
+    instance of this class represents a single field to pull from a data item.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        default: str,
+        transformation: Optional[Callable[[str], str]] = None,
+    ) -> None:
+        """
+        name: The name of the field which contains the data for display
+        default: A placeholder to use if the field is not available
+        transformation: A callable that takes and returns a string. This is used
+            to format the data field in an item
+        """
+        self.name = name
+        self.default = default
+        self.transformation = transformation
+
+
+class DisplayFields(abc.ABC):
+    """
+    An object representing how to parse lists of data into a table.
+
+    fields: A list of Fields that defines which fields get rendered into a table
+        and how they are presented
+    path_to_data_list: A key under which the data list is defined
+    """
+
+    fields: List[Field]
+    path_to_data_list: str
+    prehook: Optional[Callable[[List[Dict[str, Any]]], List[Dict[str, Any]]]] = None
+
+
+class RunListDisplayFields(DisplayFields):
+    """
+    This object defines the fields and display style of a Run listing into a
+    table.
+    """
+
+    fields = [
```

Comment:
If we decided to make this dynamic later where a user could do something like `--fields action_id, start_time, end_time` or what have you to specify which fields should be in the tabular output, we could probably do that here rather than this static list.

That'd be a follow-on feature, but if you see anything in the existing infrastructure that would make it easier to parameterize later, perhaps we try to drop it in right now.

Reply:
We could probably define (field:default:transformation) tuples for each field we want to support and let the user decide which combination of fields they want to display. Right now I don't think we need that complexity/flexibility tradeoff.